app/bundles/CoreBundle/Event/BuilderEvent.php

```diff
@@ -143,20 +143,19 @@ function ($a, $b) {
     /**
      * Adds an A/B test winner criteria option.
      *
-     * @param string $key      - a unique identifier; it is recommended that it be namespaced i.e. lead.points
-     * @param array  $criteria - can contain the following keys:
-     *                         'group'           => (required) translation string to group criteria by in the dropdown select list
-     *                         'label'           => (required) what to display in the list
-     *                         'formType'        => (optional) name of the form type SERVICE for the criteria
-     *                         'formTypeOptions' => (optional) array of options to pass to the formType service
-     *                         'callback'        => (required) callback function that will be passed the parent page or email for winner determination
-     *                         The callback function can receive the following arguments by name (via ReflectionMethod::invokeArgs())
-     *                         array $properties - values saved from the formType as defined here; keyed by page or email id in the case of
-     *                         multiple variants
-     *                         Mautic\CoreBundle\Factory\MauticFactory $factory
-     *                         Mautic\PageBundle\Entity\Page $page | Mautic\EmailBundle\Entity\Email $email (depending on the context)
-     *                         Mautic\PageBundle\Entity\Page|Mautic\EmailBundle\Entity\Email $parent
-     *                         Doctrine\Common\Collections\ArrayCollection $children
+     * @param string $key - a unique identifier; it is recommended that it be namespaced i.e. lead.points
+     * @param array{
+     *   group: string,
+     *   label: string,
+     *   event: string,
+     *   formType?: string,
+     *   formTypeOptions?: string
+     * } $criteria Can contain the following keys:
+     *  - group - (required) translation string to group criteria by in the dropdown select list
+     *  - label - (required) what to display in the list
+     *  - event - (required) event class constant that will receieve the DetermineWinnerEvent for further handling. E.g. `HelloWorldEvents::ON_DETERMINE_PLANET_VISIT_WINNER`
+     *  - formType - (optional) name of the form type SERVICE for the criteria
+     *  - formTypeOptions - (optional) array of options to pass to the formType service
```

Comment:
`callback` was removed in https://github.com/mautic/mautic/pull/8145 in favor of event listeners. Not only do we leverage PHPSTAN's static analysis now, I also replaced `callback` with `event`.